File: interface/src/main/java/xsbtpamfleti/ConsoleFactory.java
package xsbtpamfleti;

import xsbti.Logger;

public interface ConsoleFactory {
  ConsoleInterface createConsole(String[] args, String bootClasspathString,
    String classpathString, String initialCommands, String cleanupCommands,
    ClassLoader loader, String[] bindNames, Object[] bindValues,
    Logger lo);
}

File: compile-bridge/src/main/scala/xsbtpamflet/compilebridge/ConsoleFactory.scala
package xsbtpamflet

import xsbti.Logger

class ConsoleFactory extends xsbtpamfleti.ConsoleFactory {
  def createConsole(args: Array[String], bootClasspathString: String,
    classpathString: String, initialCommands: String, cleanupCommands: String,
    loader: ClassLoader, bindNames: Array[String], bindValues: Array[AnyRef],
    log: Logger): xsbtpamfleti.ConsoleInterface =
    new ConsoleInterface(args, bootClasspathString, classpathString,
      initialCommands, cleanupCommands, loader, bindNames, bindValues, log)
}
